src/containers/App/App.js

```diff
@@ -23,125 +21,96 @@ import SocketContext from '../shared/SocketContext';
 
-    /* START: Map legacy routes to new routes */
-    if (location.hash && location.pathname === '/') {
-      if (location.hash.indexOf('#/transactions/') === 0) {
-        const identifier = location.hash.split('#/transactions/')[1];
-        return <Redirect to={`/transactions/${identifier}`} />;
-      }
-      if (location.hash.indexOf('#/graph/') === 0) {
-        const identifier = location.hash.split('#/graph/')[1];
-        return <Redirect to={`/accounts/${identifier}`} />;
-      }
+  /* START: Map legacy routes to new routes */
+  if (location.hash && location.pathname === '/') {
+    if (location.hash.indexOf('#/transactions/') === 0) {
+      const identifier = location.hash.split('#/transactions/')[1];
+      return <Redirect to={`/transactions/${identifier}`} />;
     }
-    /* END: Map legacy routes to new routes */
-
-    if (location.pathname === `${urlLink}/explorer`) {
-      return <Redirect to={urlLink} />;
+    if (location.hash.indexOf('#/graph/') === 0) {
+      const identifier = location.hash.split('#/graph/')[1];
+      return <Redirect to={`/accounts/${identifier}`} />;
     }
+  }
+  /* END: Map legacy routes to new routes */
 
-    if (location.pathname === `${urlLink}/ledgers`) {
-      return <Redirect to={urlLink} />;
-    }
+  if (location.pathname === `${urlLink}/explorer`) {
+    return <Redirect to={urlLink} />;
+  }
 
-    return (
-      <div className="app">
-        <SocketContext.Provider value={this.socket}>
-          <BrowserRouter basename={rippledUrl ?? ''}>
-            <Header />
-            <div className="content">
-              <Switch>
-                <Route exact path="/" component={Ledgers} />
-                <Route exact path="/ledgers/:identifier" component={ledger} />
-                <Route exact path="/accounts/:id" component={accounts} />
-                <Route exact path="/transactions/:identifier/:tab?" component={transactions} />
-                <Route exact path="/network/:tab?" component={network} />
-                <Route exact path="/validators/:identifier/:tab?" component={validators} />
-                <Route exact path="/paystrings/:id?" component={paystrings} />
-                <Route exact path="/token/:currency.:id" component={token} />
-                {MODE === 'mainnet' && <Route exact path="/tokens" component={tokens} />}
-                <Route component={noMatch} />
-              </Switch>
-            </div>
-          </BrowserRouter>
-        </SocketContext.Provider>
-      </div>
-    );
+  if (location.pathname === `${urlLink}/ledgers`) {
+    return <Redirect to={urlLink} />;
   }
-}
+
+  return (
+    <div className="app">
+      <SocketContext.Provider value={socket}>
+        <BrowserRouter basename={rippledUrl ?? ''}>
+          <Header />
+          <div className="content">
+            <Switch>
+              <Route exact path="/" component={Ledgers} />
+              <Route exact path="/ledgers/:identifier" component={ledger} />
```

Comment:
We can address this when we upgrade to react-router.  This is how it was before.  I just moved it into a return rather than having it in a separate render function